### PR TITLE
New documentation and minor changes

### PR DESCRIPTION
**Pull request by MuellerSeb**
I have created a new documentation for the whole package and fixed some PEP8 discrepancies.

### CODE REVIEW

**Comment by LSchueler** [comment]
Wow, thanks for all the effort! The code looks a lot better and pep8ier. The documentation actually looks good now.

There are a few things I'd ask you to change:
You seem to made some copypasta:
- I found "AnaFlow" instead of "GSTools" in docs/source/conf.py
- welltestpydoc in the same file

In gstools/field/rng.py lines 55 - 64:
I think the pep8 indentations make it a lot harder to read that part

File: docs/source/conf.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
-# GeoStatTools documentation build configuration file, created by
-# sphinx-quickstart on Fri Jan  5 11:25:33 2018.
+# AnaFlow documentation build configuration file, created by
```
Comment (LSchueler) [inline docs/source/conf.py]: Copypasta

File: docs/source/conf.py [hunk 2]
```diff
@@ -123,35 +169,22 @@
 # -- Options for HTMLHelp output ------------------------------------------
 
 # Output file base name for HTML help builder.
-htmlhelp_basename = 'GeoStatToolsdoc'
+htmlhelp_basename = 'welltestpydoc'
```
Comment (LSchueler) [inline docs/source/conf.py]: Copypasta

File: gstools/field/rng.py [hunk 3]
```diff
@@ -16,56 +26,62 @@ class RNG(object):
...
         self._dim = dim
+        # set seed
+        self._seed = None
+        self._master_RNG_fct = None
+        self._master_RNG = None
         self.seed = seed
 
         self._trans_1d = lambda len_scale, r: r / len_scale
         self._trans_2d = [lambda len_scale, r, phi:
-                              r*np.cos(phi) / len_scale,
```
Comment (LSchueler) [inline gstools/field/rng.py]: I really think the pep8 standard makes things worse here

File: docs/source/conf.py (hunk 2 above)
Comment (LSchueler) [inline docs/source/conf.py]: Copypasta

File: docs/source/conf.py (hunk 1 above)
Comment (LSchueler) [inline docs/source/conf.py]: Copypasta

File: gstools/field/rng.py [hunk 4]
```diff
@@ -121,30 +138,30 @@ def _create_normal_dists(self, mode_no=None):
...
                 uniformly distributed variables
         """
         k = self.create_empty_k(mode_no)
         iid = np.empty_like(k)
 
-        for d in range(self.dim):
+        for dim_i in range(self.dim):
             rng = self._get_random_stream()
-            iid[d] = rng.uniform(0., 1., mode_no)
+            iid[dim_i] = rng.uniform(0., 1., mode_no)
```
Comment (LSchueler) [inline gstools/field/rng.py]: I think a `d` as a index variable in a short loop is perfectly fine

File: gstools/field/rng.py [hunk 5]
```diff
@@ -154,22 +171,22 @@ def gau(self, len_scale, mode_no=1000, **kwargs):
...
+            :class:`numpy.ndarray`
                 the modes
         """
         k = self.create_empty_k(mode_no)
-        for d in range(self.dim):
+        for dim_i in range(self.dim):
             rng = self._get_random_stream()
-            k[d] = rng.normal(0., 1./len_scale, mode_no)
+            k[dim_i] = rng.normal(0., 1./len_scale, mode_no)
         return k
```
Comment (LSchueler) [inline gstools/field/rng.py]: I think a `d` as a index variable in a short loop is perfectly fine

File: gstools/variogram/variogram.py [hunk 6]
```diff
@@ -61,30 +84,35 @@ def estimate_unstructured(field, bin_edges, x, y=None, z=None, sampling_size=Non
...
     return unstructured(field, bin_edges, x, y, z), bin_centres
 
-def estimate_structured(pos, field, direction='x'):
-    """Estimates the variogram of the input data on a regular grid.
+
+def estimate_structured(
+        pos,
+        field,
+        direction='x',
+):
```
Comment (LSchueler) [inline gstools/variogram/variogram.py]: What's the reasoning behind this style?

File: gstools/variogram/variogram.py [hunk 7]
```diff
@@ -61,30 +84,35 @@ def estimate_unstructured(field, bin_edges, x, y=None, z=None, sampling_size=Non
...
+    r"""Estimates the variogram of the input data on a regular grid.
```
Comment (LSchueler) [inline gstools/variogram/variogram.py]: That `r` is a great idea!

File: setup.py [hunk 8]
```diff
@@ -1,71 +1,75 @@
...
+    classifiers=CLASSIFIERS,
+    platforms=["Linux"],
+    include_package_data=True,
+    install_requires=[
+        'numpy',
+        'scipy',
+    ],
+    packages=find_packages(exclude=['tests*', 'docs*']),
+    ext_modules=EXT_MODULES,
+    include_dirs=[numpy.get_include()],
```
Comment (LSchueler) [inline setup.py]: I think the spaces here make it a lot more readible

File: tests/test_variogram_structured.py [hunk 9]
```diff
@@ -1,14 +1,21 @@
...
 """
 from __future__ import division, absolute_import, print_function
 
+import sys
 import unittest
 import numpy as np
 from gstools import variogram
 
+PY3 = sys.version_info[0] == 3
+# in python3 "long" was replaced with "int"
```
Comment (LSchueler) [inline tests/test_variogram_structured.py]: nice one!

File: gstools/field/rng.py [hunk 10]
```diff
@@ -16,56 +26,62 @@ class RNG(object):
...
+                          r*np.sin(phi) / len_scale]
         self._trans_3d = [lambda len_scale, r, theta, phi:
-                              r * np.sin(theta) * np.cos(phi) / len_scale,
+                          r * np.sin(theta) * np.cos(phi) / len_scale,
                           lambda len_scale, r, theta, phi:
-                              r * np.sin(theta) * np.sin(phi) / len_scale,
+                          r * np.sin(theta) * np.sin(phi) / len_scale,
                           lambda len_scale, r, theta, phi=None:
-                              r * np.cos(theta) / len_scale]
+                          r * np.cos(theta) / len_scale]
```
Comment (LSchueler) [inline gstools/field/rng.py]: I really think the pep8 standard makes things worse here

File: docs/source/conf.py (hunk 2 above)
Comment (MuellerSeb) [reply docs/source/conf.py]: Sorry for that.

File: docs/source/conf.py (hunk 1 above)
Comment (MuellerSeb) [reply docs/source/conf.py]: Sorry for that again. But that means even in the copied conf.py was a copy-mistake

File: gstools/field/rng.py (hunk 4 above)
Comment (MuellerSeb) [reply gstools/field/rng.py]: Ok. That's just my handwriting

File: gstools/field/rng.py (hunk 5 above)
Comment (MuellerSeb) [reply gstools/field/rng.py]: Ok. That's just my handwriting

File: gstools/field/rng.py (hunk 10 above)
Comment (MuellerSeb) [reply gstools/field/rng.py]: OK. We can just turn back to the original indentation.

File: gstools/variogram/variogram.py (hunk 6 above)
Comment (MuellerSeb) [reply gstools/variogram/variogram.py]: I think, I had something like this in mind: https://github.com/ambv/black
But in this case it's not necessary to use one line per argument. You're truly right there. Maybe we can use one line for parameters with and one for those without standard value.

File: gstools/variogram/variogram.py (hunk 7 above)
Comment (MuellerSeb) [reply gstools/variogram/variogram.py]: Yeah!

File: setup.py (hunk 8 above)
Comment (MuellerSeb) [reply setup.py]: I don't feel a big difference. But just as you wish ;)

File: tests/test_variogram_structured.py (hunk 9 above)
Comment (MuellerSeb) [reply tests/test_variogram_structured.py]: It's strange because the cython file itself isn't affected by that! I have to read about that.

**Pull request merged**